src/commands.rs

```diff
@@ -1,8 +1,34 @@
+use std::collections::HashMap;
 use tauri::command;
+use url::Url;
+
+use media_parser::{FileStreamReader, HttpStreamReader, MediaParser, Metadata};
 
 use crate::Result;
 
+/// Extract metadata from a media file (local path or URL).
+///
+/// # Arguments
+/// * `source` - Absolute path to a local file or URL of a remote media file
+/// * `headers` - Optional custom HTTP headers (only used for URLs, e.g., for authentication)
+///
+/// # Returns
+/// Metadata containing duration, timescale, and tags (title, artist, etc.)
 #[command]
-pub(crate) async fn hello(name: String) -> Result<String> {
-   Ok(format!("Hello, {}! This is the Media Parser plugin.", name))
+pub(crate) async fn get_metadata(
+   source: String,
+   headers: Option<HashMap<String, String>>,
+) -> Result<Metadata> {
+   if Url::parse(&source).is_ok() {
```

Comment:
This accepts any valid URL scheme. Consider explicitly checking for http:// or https:// to avoid confusion with file:// URLs.